File: Swift/BOJ/2000/2870.swift
//
//  2870.swift
//  Algorithm
//
//  Created by 유정주 on 2023/06/15.
//

import Foundation

let n = Int(readLine()!)!

var arr: [String] = []
for _ in 0..<n {
    let input = readLine()!.split { !$0.isNumber }.map { char -> String in
        if let num = Int(char) {
            return String(num)
        } else {
            return String(char.drop { $0 == "0" })
        }
    }
    arr.append(contentsOf: input)
}

arr.sort { ($0.count, $0) < ($1.count, $1) }

for num in arr {
    print(num)
}

File: Python/BOJ/문자열/2870.py
//
//  2870.py
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/15.
//

import sys; readLine = sys.stdin.readline
import re

n = int(readLine().rstrip())
arr = []
for _ in range(n):
    input = readLine().rstrip()
    for v in re.split('[a-z]', input):
        if v:
            arr.append(int(v))

arr.sort()

for num in arr:
    print(num)
    
